lib/index.js

```diff
@@ -96,16 +96,48 @@ function Socket (type, opts) {
             break;
     }
 
-    this.binding = nn.Socket(this.af_domain, this.protocol);
-    this.queue = [];
-
-    for(var sokopt in sol){
-        if(opts.hasOwnProperty(sokopt)) this[sokopt](opts[sokopt]);
+  this.binding = nn.Socket(this.af_domain, this.protocol);
+  this.queue = [];
+
+  /*  subscription filter control */
+  this.channels     = {};
+  this.register     = register;
+  this.chanDefault  = false;
+
+  /*  subscription handling at initialization  */
+  if (opts.hasOwnProperty('chan')) {
+    if (Array.isArray(opts.chan)) {
+      opts.chan.forEach(register.bind(this));
+    } else {
+      throw new TypeError('chan requires an Array');
     }
+  } else {
+    if (type === 'sub') {
+      this.register(''); //default topic is an empty string
+      setImmediate(function(){ this.chanDefault = true; }.bind(this));
```

Comment:
the `chanDefault` flag. 

waiting a turn of the event loop before setting `this.chanDefault` to true is a guarantee that `register()`'s check on line 122, occurs before `this.chanDefault` gets set to true.

it's excessive and we don't need `this.chanDefault`, if we set the empty string on `this.channels`, like we discussed on IRC this morning